readthedocs/projects/tasks.py

```diff
@@ -108,20 +108,46 @@ def _log(self, msg):
                          version=self.version.slug,
                          msg=msg))
 
+    # pylint: disable=arguments-differ
     def run(self, pk, version_pk=None, build_pk=None, record=True,
             docker=False, search=True, force=False, localmedia=True, **__):
-        # pylint: disable=arguments-differ
-        self.project = self.get_project(pk)
-        self.version = self.get_version(self.project, version_pk)
-        self.build = self.get_build(build_pk)
-        self.build_search = search
-        self.build_localmedia = localmedia
-        self.build_force = force
-        self.config = None
+        """
+        Run a documentation build.
 
-        setup_successful = self.run_setup(record=record)
-        if setup_successful:
-            self.run_build(record=record, docker=docker)
+        This is fully wrapped in exception handling to account for a number of failure cases.
+        """
+        try:
+            self.project = self.get_project(pk)
+            self.version = self.get_version(self.project, version_pk)
+            self.build = self.get_build(build_pk)
+            self.build_search = search
+            self.build_localmedia = localmedia
+            self.build_force = force
+            self.config = None
+
+            setup_successful = self.run_setup(record=record)
+            if setup_successful:
+                self.run_build(record=record, docker=docker)
+            failure = self.setup_env.failure or self.build_env.failure
+        except Exception as e:  # noqa
+            log.exception('Top-level build exception has been raised')
```

Comment:
This exception shouls not be logged?

Reply:
Why? Seems like we should definitely be logging it..

Reply:
@ericholscher Sorry, I could not explain it correctly.
I meant that should not this exception tracelog should be reported to user also?
@agjohnson Has explained it and I understand his point.

Reply:
So we can add extra information while sending the data to sentry so that its easier to track.
`log.exception('Top-level build exception has been raised', extra={"project": pk, "build": build_pk)`

Reply:
Yea, I think that's the plan now.